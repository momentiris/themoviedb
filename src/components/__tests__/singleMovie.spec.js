import { render } from 'react-testing-library'
import SingleMovie from '../singleMovie'
import React from 'react'

describe('components/singleMovie', () => {
  const mockedMovie = {
    details: {
      backdrop_path: '123',
      title: 'Lasse Kongo Returns',
    },
    credits: {
      cast: [1, 2, 3, 4],
    },
  }
  it('renders single movie without errors', () => {
    const { container } = render(<SingleMovie movie={mockedMovie} />)

    expect(container).toMatchSnapshot()
  })

  it('renders credits.cast', () => {
    const { container } = render(<SingleMovie movie={mockedMovie} />)
    expect(container.querySelectorAll('.cast-member').length).toEqual(4)
  })
})
