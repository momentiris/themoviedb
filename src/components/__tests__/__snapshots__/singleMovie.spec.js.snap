// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/singleMovie renders single movie without errors 1`] = `
<div>
  <article
    class="sc-bdVaJa kFFJxX"
  >
    <header
      class="sc-bxivhb jUNcIt"
    >
      Lasse Kongo Returns
    </header>
    <div
      class="sc-bwzfXH laGMxf"
    />
    <div
      class="sc-htpNat gqQVoN"
    >
      <p />
    </div>
    <h1>
      Cast and Actors
    </h1>
    <ul
      class="sc-ifAKCX kKzCkS"
    >
      <li
        class="cast-member sc-EHOje lkzxgk"
      >
        <span>
           as 
        </span>
        <span />
      </li>
      <li
        class="cast-member sc-EHOje lkzxgk"
      >
        <span>
           as 
        </span>
        <span />
      </li>
      <li
        class="cast-member sc-EHOje lkzxgk"
      >
        <span>
           as 
        </span>
        <span />
      </li>
      <li
        class="cast-member sc-EHOje lkzxgk"
      >
        <span>
           as 
        </span>
        <span />
      </li>
    </ul>
  </article>
</div>
`;
